File: Etapa3-Construcao/3.2-Stack2_Java_Spring/src/main/java/com/hotel/quartos/app/dto/QuartoListaDTO.java
package com.hotel.quartos.app.dto;

import com.hotel.quartos.domain.Disponibilidade;
import com.hotel.quartos.domain.TipoQuarto;

import java.math.BigDecimal;

// DTO de saída para listagem, contendo os campos solicitados.
public record QuartoListaDTO(
        String id,
        Integer numero,
        TipoQuarto tipo,
        BigDecimal precoHora,
        Disponibilidade disponibilidade
) {}

File: Etapa3-Construcao/3.1-Stack1_Node_TS/src/application/quartos/dtos/QuartoListaDTO.ts
import { Disponibilidade } from '../../../domain/enums/Disponibilidade.js';
import { TipoQuarto } from '../../../domain/enums/TipoQuarto.js';

export interface QuartoListaDTO {
  id: string;
  numero: number;
  tipo: TipoQuarto;
  precoHora: number;
  disponibilidade: Disponibilidade;
}
